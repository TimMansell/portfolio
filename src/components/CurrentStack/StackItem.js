import React from 'react';
import PropTypes from 'prop-types';

import SvgImage from '../SvgImage';

import styles from './CurrentStack.module.scss';

export const SkillsItem = ({ url, img, name, width }) => {
  const src = require(`./img/${img}`);

  return (
    <div className={styles.item}>
      <a href={url} target="_blank" rel="noopener noreferrer">
        <SvgImage src={src} name={name} width={width} />
      </a>
    </div>
  );
};

SkillsItem.propTypes = {
  url: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
};

export default SkillsItem;
