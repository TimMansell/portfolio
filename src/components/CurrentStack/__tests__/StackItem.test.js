import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import StackItem from '../StackItem';

const props = {
  url: '',
  img: 'babel.svg',
  name: '',
  width: ''
};

describe('StackItem', () => {
  it('should render my component', () => {
    const wrapper = shallow(<StackItem {...props} />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<StackItem {...props} />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
