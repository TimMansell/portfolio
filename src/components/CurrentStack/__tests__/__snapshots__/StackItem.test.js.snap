// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackItem should match snapshot 1`] = `
<div
  className="item"
>
  <a
    href=""
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt=""
      className="img"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "babel.svg",
        }
      }
      title=""
      width=""
    />
  </a>
</div>
`;
